pkg/webhook/parser_integration_test.go

```diff
@@ -80,6 +82,23 @@ var _ = Describe("Webhook Generation From Parsing to CustomResourceDefinition",
 		}
 		assertSame(actualMutating, expectedMutating)
 		assertSame(actualValidating, expectedValidating)
+
+		By("loading the generated v1 YAML")
+		actualFile, err = ioutil.ReadFile(path.Join(outputDir, "manifests.v1.yaml"))
+		Expect(err).NotTo(HaveOccurred())
+		actualMutating, actualValidating = unmarshalBoth(actualFile)
+
+		By("loading the desired v1 YAML")
+		expectedFile, err = ioutil.ReadFile("manifests.v1.yaml")
+		Expect(err).NotTo(HaveOccurred())
+		expectedMutating, expectedValidating = unmarshalBoth(expectedFile)
+
+		By("comparing the two")
+		assertSame = func(actual, expected interface{}) {
+			ExpectWithOffset(1, actual).To(Equal(expected), "type not as expected, check pkg/webhook/testdata/README.md for more details.\n\nDiff:\n\n%s", cmp.Diff(actual, expected))
+		}
+		assertSame(actualMutating, expectedMutating)
```

Comment:
Would it maybe make sense to add an `envtest` based test to verify that the kube apiserver actualy accepts these?

Reply:
Yes, sure, do we have any examples for this in our codebase?

Reply:
Oh, I see. I can use the `Paths` in the `WebhookInstallOptions` https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/envtest/webhook.go#L44 to register to apiserver, and then do `env.Start()` to start the apiserver to see if there are any errors
@alvaroaleman Is this what you meant?

Reply:
Either that or do a `Create` after its started, that way the error might be more helpful, if any.

Reply:
Working on this :)

Reply:
I would be ok with adding envtest in a follow-up.

Reply:
@alvaroaleman I have tried to add the `envtest` based test, finally, it works but still some issues around it:
1. We need to add `sigs.k8s.io/controller-runtime` as a dependency. (If we require v0.6.2, then we need to bump k8s version)
2. We need to upgrade the k8s version in test.sh: https://github.com/kubernetes-sigs/controller-tools/blob/master/test.sh#L41 v1.11.0 is too old and `v1` version WebhookConfiguration is not available, and will receive something like this:
```
  Unexpected error:
      <*meta.NoKindMatchError | 0xc0009c5100>: {
          GroupKind: {
              Group: "admissionregistration.k8s.io",
              Kind: "MutatingWebhookConfiguration",
          },
          SearchedVersions: ["v1"],
      }
      no matches for kind "MutatingWebhookConfiguration" in version "admissionregistration.k8s.io/v1"
```
Do you think it is still necessary to do this in this PR? Or we regard this as a follow-up, and I can make small PRs to bump k8s version, upgrade k8s version in the test.sh, and add test?

Reply:
Follow-Up is fine